api/controller/common.proto

```diff
@@ -33,3 +33,25 @@ message ComputeResources {
   // Memory in bytes.
   int64 memory_bytes = 2;
 }
+
+// PMMParams represents params related to pmm. If nil do not turn on monitoring.
+message PMMParams {
+  // PMM server public address.
+  string public_address = 1 [
+    (validator.field) = {
+      msg_exists: true
```

Comment:
I cloud be wrong, but maybe  we should check if `public_address`, `login` and `password` not empty when PMMParams not nil? 

https://github.com/mwitkow/go-proto-validators/issues/27#issuecomment-302228034

Reply:
it actually works this way

Reply:
Just checked, I don't see string validation, Validation method is empty.

https://github.com/percona-platform/dbaas-api/pull/121/files#diff-0134b6666666dd3974e2e96dd7106e2efe5c3207dfdd2a555e4b860491250975R32

Also I checked locally, here is my diff when changing msg_exists to string_not_empty

```
diff --git a/api/controller/common.proto b/api/controller/common.proto
index d613f25..80fbb55 100644
--- a/api/controller/common.proto
+++ b/api/controller/common.proto
@@ -39,19 +39,19 @@ message PMMParams {
   // PMM server public address.
   string public_address = 1 [
     (validator.field) = {
-      msg_exists: true
+      string_not_empty: true
     }
   ];
   // PMM server admin login.
   string login = 2 [
     (validator.field) = {
-      msg_exists: true
+      string_not_empty: true
     }
   ];
   // PMM server admin password.
   string password = 3 [
     (validator.field) = {
-      msg_exists: true
+      string_not_empty: true
     }
   ];
 }
diff --git a/gen/controller/common.pb.go b/gen/controller/common.pb.go
index ccd753d..e4f6c3b 100644
--- a/gen/controller/common.pb.go
+++ b/gen/controller/common.pb.go
@@ -296,12 +296,12 @@ var file_controller_common_proto_rawDesc = []byte{
 	0x02, 0x20, 0x01, 0x28, 0x03, 0x52, 0x0b, 0x6d, 0x65, 0x6d, 0x6f, 0x72, 0x79, 0x42, 0x79, 0x74,
 	0x65, 0x73, 0x22, 0x7c, 0x0a, 0x09, 0x50, 0x4d, 0x4d, 0x50, 0x61, 0x72, 0x61, 0x6d, 0x73, 0x12,
 	0x2d, 0x0a, 0x0e, 0x70, 0x75, 0x62, 0x6c, 0x69, 0x63, 0x5f, 0x61, 0x64, 0x64, 0x72, 0x65, 0x73,
-	0x73, 0x18, 0x01, 0x20, 0x01, 0x28, 0x09, 0x42, 0x06, 0xe2, 0xdf, 0x1f, 0x02, 0x20, 0x01, 0x52,
+	0x73, 0x18, 0x01, 0x20, 0x01, 0x28, 0x09, 0x42, 0x06, 0xe2, 0xdf, 0x1f, 0x02, 0x58, 0x01, 0x52,
 	0x0d, 0x70, 0x75, 0x62, 0x6c, 0x69, 0x63, 0x41, 0x64, 0x64, 0x72, 0x65, 0x73, 0x73, 0x12, 0x1c,
 	0x0a, 0x05, 0x6c, 0x6f, 0x67, 0x69, 0x6e, 0x18, 0x02, 0x20, 0x01, 0x28, 0x09, 0x42, 0x06, 0xe2,
-	0xdf, 0x1f, 0x02, 0x20, 0x01, 0x52, 0x05, 0x6c, 0x6f, 0x67, 0x69, 0x6e, 0x12, 0x22, 0x0a, 0x08,
+	0xdf, 0x1f, 0x02, 0x58, 0x01, 0x52, 0x05, 0x6c, 0x6f, 0x67, 0x69, 0x6e, 0x12, 0x22, 0x0a, 0x08,
 	0x70, 0x61, 0x73, 0x73, 0x77, 0x6f, 0x72, 0x64, 0x18, 0x03, 0x20, 0x01, 0x28, 0x09, 0x42, 0x06,
-	0xe2, 0xdf, 0x1f, 0x02, 0x20, 0x01, 0x52, 0x08, 0x70, 0x61, 0x73, 0x73, 0x77, 0x6f, 0x72, 0x64,
+	0xe2, 0xdf, 0x1f, 0x02, 0x58, 0x01, 0x52, 0x08, 0x70, 0x61, 0x73, 0x73, 0x77, 0x6f, 0x72, 0x64,
 	0x42, 0x1e, 0x5a, 0x1c, 0x63, 0x6f, 0x6e, 0x74, 0x72, 0x6f, 0x6c, 0x6c, 0x65, 0x72, 0x3b, 0x63,
 	0x6f, 0x6e, 0x74, 0x72, 0x6f, 0x6c, 0x6c, 0x65, 0x72, 0x76, 0x31, 0x62, 0x65, 0x74, 0x61, 0x31,
 	0x62, 0x06, 0x70, 0x72, 0x6f, 0x74, 0x6f, 0x33,
diff --git a/gen/controller/common.validator.pb.go b/gen/controller/common.validator.pb.go
index 8d06935..06d18fb 100644
--- a/gen/controller/common.validator.pb.go
+++ b/gen/controller/common.validator.pb.go
@@ -30,5 +30,14 @@ func (this *ComputeResources) Validate() error {
 	return nil
 }
 func (this *PMMParams) Validate() error {
+	if this.PublicAddress == "" {
+		return github_com_mwitkow_go_proto_validators.FieldError("PublicAddress", fmt.Errorf(`value '%v' must not be an empty string`, this.PublicAddress))
+	}
+	if this.Login == "" {
+		return github_com_mwitkow_go_proto_validators.FieldError("Login", fmt.Errorf(`value '%v' must not be an empty string`, this.Login))
+	}
+	if this.Password == "" {
+		return github_com_mwitkow_go_proto_validators.FieldError("Password", fmt.Errorf(`value '%v' must not be an empty string`, this.Password))
+	}
 	return nil
 }
```

Reply:
👍🏼 will replace with `string_not_empty`